### PR TITLE
docs(keda): Introduce helm-docs for documentation

### LINKED ISSUE #444: Use helm-docs to document params in README.md
I'd like to introduce https://github.com/norwoodj/helm-docs for documenting the possible parameters.

### Use-Case

It's hard to follow if every param is up-to-date on a PR. We at argo-helm use helm-docs to mitigate this.

### Specification

- Update values.yaml with helm-docs comment above each key/param
- Introduce helm-docs
- Update contributor docs about helm-docs

### PR DESCRIPTION
**Pull request by mkilchhofer**
Resolves #444 

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)*

### CODE REVIEW

**Comment by mkilchhofer** [comment]
Rebased after merge of PR #257

**Comment by tomkerkhove** [comment]
Sorry for late reply but would you mind how this is better than what we have please? This is new to me.

**Comment by mkilchhofer** [comment]
Today it depends on the contributor to remember to update the README after introducing/changing new variables.

Helm-docs seems to be some kind of standard inside the helms community. This is why I opened the issue for talking about this before spending time on the implementation :-)

**Review by JorTurFer — APPROVED** [review]

**Comment by JorTurFer** [comment]
Thanks for this awesome contribution ❤️  This tool is fascinating
WDYT @kedacore/keda-helm-contributors ?

**Comment by mkilchhofer** [comment]
Rebased the PR again :-)

**Review by tomkerkhove — APPROVED** [review]
LGTM, thanks a ton!
